src/app/services/game-state.service.ts

```diff
@@ -164,230 +90,83 @@ export class GameStateService {
     return this.isMyTurn$().pipe(take(1));
   }
 
-  getMyLoginName(): string {
-    return this.colyseus.myLoginName;
+  getMyLoginNameOnce$(): Observable<string> {
+    return this.colyseus.myLoginName$.pipe(take(1));
   }
 
-  getMyFigureId(): number {
-    const p: Player = this.getByLoginName(this.colyseus.myLoginName);
-    return p.figureId;
+  findInPlayerListOnce$(f: (p: Player) => boolean): Observable<Player | undefined> {
+    return this.observableState.playerList$
+      .pipe(take(1))
```

Comment:
You can write this pipe as:
```
.pipe(
    take(1),
    map((...) => foo)
)
```
This can make the code less cluttered but it is subjective if this is better